drivers/led/pca963x.c

```diff
@@ -0,0 +1,265 @@
+	    value > led_data->max_brightness) {
+		return -EINVAL;
+	}
+
+	/* Set the LED brightness value */
+	val = (value * 255U) / led_data->max_brightness;
+	if (i2c_reg_write_byte_dt(&config->i2c,
+			       config->pwm_base + led,
+			       val)) {
+		LOG_ERR("LED reg write failed");
+		return -EIO;
+	}
+
+	/* Set the LED driver to be controlled through its PWMx register. */
+	if (i2c_reg_update_byte_dt(&config->i2c,
+				config->ledout + (led / 4),
+				PCA963X_MASK << ((led % 4) << 1),
+				PCA963X_LED_PWM << ((led % 4) << 1))) {
+		LOG_ERR("LED reg update failed");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static inline int pca963x_led_on(const struct device *dev, uint32_t led)
+{
+	const struct pca963x_config *config = dev->config;
+
+	/* Set LED state to ON */
+	if (i2c_reg_update_byte_dt(&config->i2c,
+				config->ledout + (led / 4),
+				PCA963X_MASK << ((led % 4) << 1),
+				PCA963X_LED_ON << ((led % 4) << 1))) {
+		LOG_ERR("LED reg update failed");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static inline int pca963x_led_off(const struct device *dev, uint32_t led)
+{
+	const struct pca963x_config *config = dev->config;
+
+	/* Set LED state to OFF */
+	if (i2c_reg_update_byte_dt(&config->i2c,
+				config->ledout + (led / 4),
+				PCA963X_MASK << ((led % 4) << 1),
+				PCA963X_LED_OFF)) {
+		LOG_ERR("LED reg update failed");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int pca963x_led_init(const struct device *dev)
+{
+	const struct pca963x_config *config = dev->config;
+
+	if (!device_is_ready(config->i2c.bus)) {
+		LOG_ERR("I2C bus is not ready");
+		return -ENODEV;
+	}
+
+	/* Take the LED driver out from Sleep mode. */
+	if (i2c_reg_update_byte_dt(&config->i2c,
+				PCA963X_MODE1,
+				PCA963X_MODE1_SLEEP,
+				~PCA963X_MODE1_SLEEP)) {
+		LOG_ERR("LED reg update failed");
+		return -EIO;
+	}
+
+	/* If there is external driver, set invrt bit */
+	if (config->external_driver) {
+		if (i2c_reg_update_byte_dt(&config->i2c,
+					PCA963X_MODE2,
+					PCA963X_MODE2_INVRT,
+					PCA963X_MODE2_INVRT)) {
+			LOG_ERR("LED reg update failed");
+			return -EIO;
+		}
+	}
+
+	return 0;
+}
+
+static const struct led_driver_api pca963x_led_api = {
+	.blink = pca963x_led_blink,
+	.set_brightness = pca963x_led_set_brightness,
+	.on = pca963x_led_on,
+	.off = pca963x_led_off,
+};
+
+#define PCA963x_DEVICE(node_id, prefix, regs, params)						\
+	static const struct pca963x_config _CONCAT(prefix ## _config_, DT_DEP_ORD(node_id)) = {	\
+		.i2c = I2C_DT_SPEC_GET(node_id),						\
+		.led_data = {									\
+			.min_period = params ## _MIN_PERIOD,					\
+			.max_period = params ## _MAX_PERIOD,					\
+			.min_brightness = params ## _MIN_BRIGHTNESS,				\
+			.max_brightness = params ## _MAX_BRIGHTNESS,				\
+		},										\
+		.pwm_base = regs ## _PWM_BASE,							\
+		.grppwm = regs ## _GRPPWM,							\
+		.grpfreq = regs ## _GRPFREQ,							\
+		.ledout = regs ## _LEDOUT,							\
+		.external_driver = DT_PROP_OR(node_id, external_driver, false),			\
```

Comment:
Just use DT_PROP(node_id, external_driver). DT_PROP() does the right thing for boolean properties. It knows that a missing boolean property is `false`.